### PR TITLE
fix(ci): release workflow for preview releases and pypi publish

### PR DESCRIPTION
**Pull request by BeArchiTek**
This PR adds pypi publishing of infrahub-server during a release.

It also fixes issues we had with the current release workflow: it did not take account of all syntax cases for the version, especially when releasing a preview release. This PR switches the workflow to leverage the Github release event, check if it's a github pre-release, validates if the (pre)release matches the python version.

It also fixes an edge case where the "latest" docker tag would be assigned to older versions, we now make use of the "latest" github release flag.

### CODE REVIEW

File: .github/workflows/publish-pypi.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+          installer-parallel: true
+
+      - name: "Check out repository code"
+        uses: "actions/checkout@v4"
+
+      - name: "Cache poetry venv"
+        uses: "actions/cache@v4"
+        id: "cached-poetry-dependencies"
+        with:
+          path: "./infrahub_sync/.venv"
```
Comment (ogenstad) [inline .github/workflows/publish-pypi.yml]: I'm guessing this is copied from another repo.

Comment (BeArchiTek) [reply .github/workflows/publish-pypi.yml]: spotted on Patrick :p 
It was a kickstart for @fatih-acar  to continue but yeah bad copy-paste

**Comment by fatih-acar** [comment]
Edited the PR description with new changes around the release workflow.
cc @lykinsbd

**Review by BeArchiTek — COMMENTED** [review]
LGTM

**Review by wartraxx51 — APPROVED** [review]

**Review by lykinsbd — APPROVED** [review]

**Pull request merged**